jobs/release-monitoring/release-monitoring-3scale/Jenkinsfile

```diff
@@ -0,0 +1,202 @@
+    def latestPatch = latest[2].tokenize('-')[0] as Integer
+
+    def previousVer = 0
+    def currentVer = 0
+    for (i = 0; i <= 2; i++) {
+        if (i > 0) {
+            previousVer = i - 1
+        }
+        currentVer = i
+
+        def previousDiff = current[previousVer] == latest[previousVer]
+        if (i == 0) {
+            previousDiff = true
+        }
+        def currentDiff =  latest[currentVer] > current[currentVer]
+        if (previousDiff && currentDiff) {
+            return true
+        }
+    }
+
+    return false
+}
+
+def installationGitUrl = params.installationGitUrl
+def installationGitRef = params.installationGitRef
+def manifestVar = params.manifestVar
+def projectOrg = params.projectOrg
+def projectRepo = params.projectRepo
+def githubToken = params.githubToken
+def githubCredentialsID = params.credentialId
+def nextBranch = params.installationProductBranch
+def clusterConfigCredentialsID = params.clusterConfigCredentialsID
+def dryRun = params.dryRun
+def installJobName = params.installJobName
+def testJobName = params.testJobName
+def uninstallJobName = params.uninstallJobName
+
+node {
+    stage('Fetch Installation Repo') {
+        println("Fetch Installation Repo")
+
+        cleanWs()
+        dir('installation') {
+            checkoutGitRepo(installationGitUrl, installationGitRef, githubCredentialsID)
+            releaseConfig = readYaml file: "./evals/inventories/group_vars/all/manifest.yaml"
+            releaseVersion = releaseConfig[manifestVar]
+            componentRelease = releaseVersion
+        }
+    }
+
+    stage('Fetch latest release') {
+        def (code, data) = getLatestRelease(projectOrg, projectRepo, githubToken)
+        if (code != 200) {
+            print("Failed to retrieved latest release: ${data}")
+            currentBuild.result = "FAILURE"
+            return
+        }
+        def tagName = data.tag_name
+        latestRelease = tagName
+    }
+
+    stage('Diff Release Versions') {
+        if (latestRelease == "" || componentRelease == "") {
+            currentBuild.result = "FAILURE"
+            return
+        }
+
+        if (!hasNewRelease(componentRelease, latestRelease)) {
+            print('[DEBUG] No new releases available')
+            currentBuild.result = "ABORTED"
+            return
+        }
+
+        print('[DEBUG] Found new release')
+
+        releaseConfig[manifestVar] = latestRelease
+        dir('installation') {
+            sshagent([githubCredentialsID]) {
+                sh 'git config --global user.name "Automatron"'
+                sh 'git config --global user.email "github.eng@feedhenry.com"'
+                def branchCode = sh script: "git checkout -b ${nextBranch}", returnStatus: true
+                if (branchCode  != 0) {
+                    sh "git checkout ${nextBranch}"
+                }
+                sh "git fetch origin && git reset --hard origin/${nextBranch} || true"
```

Comment:
What's the point of this?

Reply:
to reset to the latest next branch commit if we already have one (in case we have more than one product release before merging it back to master)

Reply:
OK, so you could do it in the if  above? and expect it to work? Should you not also be rebasing master into this branch at some point here?